spark/v3.3/spark-extensions/src/main/antlr/org.apache.spark.sql.catalyst.parser.extensions/IcebergSqlExtensions.g4

```diff
@@ -82,6 +82,13 @@ snapshotRetentionClause
     | WITH SNAPSHOT RETENTION numSnapshots SNAPSHOTS snapshotRetain snapshotRetainTimeUnit
     ;
 
+createReplaceBranchClause
+    : (CREATE | REPLACE) BRANCH identifier branchOptions
+    ;
```

Comment:
and if desired in the future we should be able to extend this to CREATE OR REPLACE and make minimal changes in the AST/exec to handle this case.

Reply:
I am okay if we just change this PR to just support all combinations. Based on https://docs.databricks.com/sql/language-manual/sql-ref-syntax-ddl-create-table-using.html, we should do:

`{ { [CREATE OR] REPLACE BRANCH | CREATE BRANCH [ IF NOT EXISTS ] }`